File: src/backend/app/models/table.py
from pydantic import BaseModel
from typing import Optional
from app.models.database import ConnectionInfo


class TableListRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    schema_name: str


class TableListResponse(BaseModel):
    tables: list[str]


class CreateTableRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    ddl: str


class DeleteTableRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    schema_name: str
    table_name: str


class ColumnInfo(BaseModel):
    column_name: str
    data_type: str
    is_nullable: str
    column_default: Optional[str] = None


class IndexInfo(BaseModel):
    index_name: str
    column_name: str
    is_unique: bool


class TableStructureRequest(BaseModel):
    connection: ConnectionInfo
    database_name: str
    schema_name: str
    table_name: str


class TableStructureResponse(BaseModel):
    columns: list[ColumnInfo]
    indexes: list[IndexInfo]

File: frontend/src/types/table.ts
export interface ColumnInfo {
  column_name: string;
  data_type: string;
  is_nullable: string;
  column_default: string | null;
}

export interface IndexInfo {
  index_name: string;
  column_name: string;
  is_unique: boolean;
}

export interface TableStructure {
  columns: ColumnInfo[];
  indexes: IndexInfo[];
}
